### PR TITLE
Fix result shape of tf.tensordot unknown when axes is an integer number

### PR DESCRIPTION
**Pull request by lspvic**
#8452 add the function "Tensordot partial shape inference", solves the problem #6682.
However, the shape of result of `tensordot` is still `<unknown>` when `axes` is an integer N, which is in common use.
For example, 
```
a = tf.placeholder('float32', shape=[None, 100])
b = tf.placeholder('float32', shape=[100, 300])
```
set `axes=1`,
```
result_tensordot = tf.tensordot(a, b, axes=1)
result_tensordot.get_shape()  # TensorShape(None)
result_tensordot.get_shape().as_list()  # Error
```
The equivalent `axes=[[1], [0]]` behaves correctly,
```
result_tensordot = tf.tensordot(a, b, axes=[[1], [0]])
result_tensordot.get_shape()  # TensorShape([Dimension(None), Dimension(300)])
result_tensordot.get_shape().as_list()  # [None, 300]
```
The simplified is more common and the partial shape should be inferred correctly.
This PR solves the problem.

### CODE REVIEW

**Review by rmlarsen — CHANGES_REQUESTED** [review]

File: tensorflow/python/ops/math_ops.py [hunk 1]
```diff
@@ -2768,7 +2768,8 @@ def _tensordot_axes(a, axes):
       if axes < 1:
         raise ValueError("'axes' must be at least 1.")
       if a_shape.ndims is not None:
-        return range(a_shape.ndims - axes, a_shape.ndims), range(axes)
+        return (list(xrange(a_shape.ndims - axes, a_shape.ndims)),
```
[the remark this answers is not in the document]
Comment (lspvic) [reply tensorflow/python/ops/math_ops.py]: I find that there's no test units for math_ops.tensordot. I add a simple case for just this case to test the output shape inference when `axes` is integer numbers.

**Comment by lspvic** [comment]
@rmlarsen 
I also found a problem with the previous check code on `axes`, which needs `axes>= 1`. 
```
      if axes < 1:
        raise ValueError("'axes' must be at least 1.")
```
However, `axes=0` is also correct in numpy meaning tensor outer product a ⊗ b in [np.tensordot](https://docs.scipy.org/doc/numpy-1.13.0/reference/generated/numpy.tensordot.html). Moreover,
the equivalent `axes=[[], []]` is  correctly behaved in current tensorflow code. So, I change the code 
to
```
      if axes < 0:
        raise ValueError("'axes' must be at least 0.")
```
And also, checks that `axes` less equal than the ndim of tensor's shape for meaning of the last or first N dimensions.


Another doubt on that `np.tensordot` has a default `axes=2`  for tensor double contraction, is it necessary to change the `tf.tensordot` behavior more like numpy?

**Comment by rmlarsen** [comment]
@lspvic Thanks for fixing the additional bug. I would prefer to keep the existing default for now, since changing it might break existing code. We don't have to match Numpy in all casees, but I agree that it might make sense here. Would you be willing to make that single change in a separate PR?

**Comment by lspvic** [comment]
@rmlarsen Thanks for your advise. Now this PR do the following things and should be compatible with previous codes:
1. Fix the bug that scalar `axes` will not output correct tensor shape. Test is add for `tensordot(a, b, 1)` on `test_partial_shape_inference`;
2. `axes` can be set to 0 now while previous codes just disable 0. Actually `axes=0` is supported before with the equivalent format `axes=[[[], []]`. Tests for `axes=0` and `axes=[[], []]` are added and they both works well.
3. Additional check on that `axes` scalar values must be less or equal than either of input tensors. Test case is also added.

**Comment by rmlarsen** [comment]
@lspvic excellent, thank you for this contribution!

**Review by rmlarsen — APPROVED** [review]

**Pull request merged**